src/server/src/test/java/io/cassandrareaper/resources/view/NodesStatusTest.java

```diff
@@ -221,6 +227,89 @@ private void testParseEndpoint22StatusString(String sourceNode,
     assertTrue(nodesStatus.endpointStates.get(0).endpointNames.contains(afterSlash(endpoint3)));
   }
 
+
+  private void testParseEndpoint40StatusString(String sourceNode,
+                                               String endpoint1, String endpoint2, String endpoint3) {
+    Map<String, String> simpleStates = Maps.newHashMap();
+
+    String endpointsStatusString = endpoint1 + "\n"
+            + "  generation:1496849190\n"
+            + "  heartbeat:1231900\n"
+            + "  STATUS_WITH_PORT:14:NORMAL,-9223372036854775808\n"
+            + "  LOAD:1231851:4043215.0\n"
+            + "  SCHEMA:10:2de0af6a-bf86-38e0-b62b-474ff6aefb51\n"
+            + "  DC:6:datacenter1\n"
+            + "  RACK:8:rack1\n"
+            + "  RELEASE_VERSION:4:3.0.8\n"
+            + "  RPC_ADDRESS:3:127.0.0.1\n"
+            + "  SEVERITY:1231899:0.0\n"
+            + "  NET_VERSION:1:10\n"
+            + "  HOST_ID:2:f091f82b-ce2c-40ee-b30c-6e761e94e821\n"
+            + "  RPC_READY:16:true\n"
+            + "  TOKENS:13:<hidden>\n"
+            + endpoint2 + "\n"
+            + "  generation:1497347537\n"
+            + "  heartbeat:27077\n"
+            + "  STATUS:14:NORMAL,-3074457345618258603\n"
+            + "  LOAD:1231851:3988763.0\n"
+            + "  SCHEMA:10:2de0af6a-bf86-38e0-b62b-474ff6aefb51\n"
+            + "  DC:6:datacenter2\n"
+            + "  RACK:8:rack2\n"
+            + "  RELEASE_VERSION:4:3.0.8\n"
+            + "  RPC_ADDRESS:3:127.0.0.2\n"
+            + "  SEVERITY:1231899:0.0\n"
+            + "  NET_VERSION:1:10\n"
+            + "  STATUS_WITH_PORT:14:NORMAL,-3074457345618258603\n"
```

Comment:
Did you intend to have `STATUS` and `STATUS_WITH_PORT` both in the status string here?

Reply:
Is it guaranteed that the statuses would/should match? If not, I'm assuming STATUS_WITH_PORT wins, according to the ordering of the status matchers.

Reply:
yep, so far I observed that they were strictly identical (which is expected I guess). This change comes from the fact that 4.0 allows to run several Cassandra instances on the same server. They're then identified by host:port instead of just host in 3.11.

Reply:
:+1: